### DISCUSSION TITLE
developer.mend.io and private repos: security considerations

### DISCUSSION THREAD

**Discussion by arseny-zinchenko** [renovatebot/renovate#29327] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi.
I'm just started using Renovate in our GitHub org, and a bit concerned about the security: am I right, that it clones the code to the developer.mend.io?
If so, what about private code in private repositories, which must not be shared anywhere?
I tried to find a way to disable the developer.mend.io at all, but can't see such an option.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Renovate needs to clone code in order to scan it for dependencies. Code is deleted at the end of every run and not stored persistently. If you don't trust Mend to do that, you can self host Renovate.

**Reply by HonkingGoose** [reply]
@arseny-zinchenko you will want to read the [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/) as well.
